kafka/src/test/java/org/jboss/aerogear/unifiedpush/kafka/consumers/InstallationMetricsConsumerTest.java

```diff
@@ -0,0 +1,109 @@
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.kafka.consumers;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import org.jboss.aerogear.unifiedpush.kafka.utils.MockProviders;
+import org.jboss.aerogear.unifiedpush.kafka.BasicKafkaTest;
+import org.jboss.aerogear.unifiedpush.kafka.KafkaClusterConfig;
+import org.jboss.aerogear.unifiedpush.kafka.MessageConsumedEvent;
+import org.jboss.aerogear.unifiedpush.rest.AbstractBaseEndpoint;
+import org.jboss.aerogear.unifiedpush.rest.registry.installations.InstallationRegistrationEndpoint;
+import org.jboss.aerogear.unifiedpush.service.metrics.PushMessageMetricsService;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import net.wessendorf.kafka.impl.DelegationKafkaConsumer;
+
+/**
+ * Test cases for {@link InstallationMetricsConsumer#consume(String)} method.
+ */
+@RunWith(Arquillian.class)
+public class InstallationMetricsConsumerTest extends BasicKafkaTest {
+
+    private boolean isMethodCalled = Boolean.FALSE;
+    private static CountDownLatch countDownLatch;
+
+    @Inject
+    InstallationRegistrationEndpoint installationRegistrationEndpoint;
+
+    @Deployment
+    public static JavaArchive createDeployment() {
+        // initialize system variables that will be used by kafka cluster
+        init();
+
+        // deploy a jar file
+        return ShrinkWrap.create(JavaArchive.class)
+                // add Kafka configurations
+                .addClass(KafkaClusterConfig.class)
+                // add the consumer that is tested
+                .addClass(InstallationMetricsKafkaConsumer.class)
+                // add a container for all mock providers
+                .addPackage(MockProviders.class.getPackage())
+                // needed to the cdi library
+                .addClass(DelegationKafkaConsumer.class).addClass(InstallationRegistrationEndpoint.class)
+                .addClass(AbstractBaseEndpoint.class).addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Test
+    public void consumeTest() throws InterruptedException {
```

Comment:
This test technically checks that the `updateAnalytics` method is called as expected ~ while that _should_ tell us that the message was consumed correctly, that's not necessarily the case so it might be nice to have another test which confirms this, similar to the one in the producer test?